server/src/test/java/io/crate/execution/engine/collect/files/FileReadingIteratorTest.java

```diff
@@ -305,6 +302,70 @@ public String readLine() throws IOException {
         tester.verifyResultAndEdgeCaseBehaviour(expectedResult);
     }
 
+    @Test
+    public void test_skipping_csv_headers_and_rows_combined_with_retry_logic() throws Exception {
+        final int skipNumLines = 2;
+        ArrayList<String> fileUris = new ArrayList<>();
+
+        var tempFilePath1 = createTempFile("tempfile1", ".csv");
+        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tempFilePath1.toFile()),
+                                                                StandardCharsets.UTF_8)) {
+            writer.write("id\n");
+            writer.write("1\n2\n3\n4\n5\n");
+        }
+        fileUris.add(tempFilePath1.toUri().toString());
```

Comment:
Could shorten this a bit:


```suggestion
        Path tempFile = createTempFile("tempfile1", ".csv");
        Files.write(tempFile, List.of("id", "1", "2", "3", "4", "5"));
        List<String> fileUris = List.of(tempFile.toUri().toString());
```

Reply:
Also applied, Thanks.